src/worker.cc

```diff
@@ -51,6 +51,7 @@ const std::unordered_map<unsigned, valhalla::valhalla_exception_t> error_codes{
     {100, {100, "Failed to parse json request", 400, HTTP_400, OSRM_INVALID_URL, "json_parse_failed"}},
```

Comment:
all of the annoying changes to make json parsing also care about a potentially already provided protobuf object are in this file. i mentioned them in the pr description so i wont go into detail here.